chromadb/auth/basic_authn/__init__.py

```diff
@@ -0,0 +1,130 @@
+import base64
+import importlib
+import logging
+
+from fastapi import HTTPException
+from overrides import override
+from pydantic import SecretStr
+
+from chromadb.auth import (
+    UserIdentity,
+    ServerAuthenticationProvider,
+    ClientAuthProvider,
+    ClientAuthHeaders,
+)
+from chromadb.config import System
+from chromadb.telemetry.opentelemetry import (
+    OpenTelemetryGranularity,
+    trace_method,
+)
+from starlette.datastructures import Headers
+from typing import Dict
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["BasicAuthenticationServerProvider", "BasicAuthClientProvider"]
+
+
+class BasicAuthClientProvider(ClientAuthProvider):
+    """
+    Client auth provider for basic auth. The credentials are passed as a
+    base64-encoded string in the Authorization header prepended with "Basic ".
+    """
+    def __init__(self, system: System) -> None:
+        super().__init__(system)
+        self._settings = system.settings
+        system.settings.require("chroma_client_auth_credentials")
+        self._creds = SecretStr(
+            str(system.settings.chroma_client_auth_credentials)
+        )
+
+    @override
+    def authenticate(self) -> ClientAuthHeaders:
+        # We do this goofy encode-decode dance to make the type checker happy.
```

Comment:
nit: should we just ignore instead

Reply:
Turns out this comment was wrong -- we do need to do this to ensure it's correctly formatted. Removed the comment.